File: src/puck/geometry/wrapAngle.ts
import { puckGeometry } from "./puckGeometry";

/* Wrap an angle back into (−π, π]: the shortest path between two measurements. */
export function wrapAngle(a: number): number {
    const g = puckGeometry.exports;
    if (g) return g.wrap_angle_rad(a);
    while (a > Math.PI) a -= Math.PI * 2;
    while (a < -Math.PI) a += Math.PI * 2;
    return a;
}

File: src/wasm/puck-geometry/src/wrap_angle.rs
//! Wrap an angle back into (−π, π].

/// Wrap an angle back into (−π, π]: the shortest path between two measurements.
pub fn wrap_angle(mut a: f64) -> f64 {
    let tau = std::f64::consts::PI * 2.0;
    while a > std::f64::consts::PI {
        a -= tau;
    }
    while a < -std::f64::consts::PI {
        a += tau;
    }
    a
}
